src/Confluent.Kafka/Consumer.cs

```diff
@@ -645,13 +659,8 @@ public void Assign(IEnumerable<TopicPartition> partitions)
         public void Unassign()
             => kafkaHandle.Assign(null);
 
-        /// <summary>
-        ///     Manually consume message or triggers events.
-        ///
-        ///     Will invoke events for OnPartitionsAssigned/Revoked,
-        ///     OnOffsetCommit, etc. on the calling thread.
-        /// </summary>
-        public bool Consume(out Message message, int millisecondsTimeout)
+
+        internal bool ConsumeImpl(out Message message, int millisecondsTimeout)
```

Comment:
I find myself in conflict with the Microsoft recommendations here. 

First I want to defend the use of the out parameter in this scenario - doing it this way here results in both terser usage patterns and avoids making `Message` a class or wrapping it in `Nullable`.

Second, ordering as I have seems more natural (timeout is optional - put last). In fact, if we had have used a default parameter value for the timeout, it would have been required to be specified last. 

I'm quite happy to argue on this one out more, but at this point I feel like I want to leave as is.

side question: @tomasdeml - would love to know if you have any opinion on whether default parameter values (like we use in ProduceAsync methods) should be avoided? BCL doesn't use them as far as I can tell, but we came to the conclusion that it was ok.